packages/web-components/src/wallet-connection/walletConnection.js

```diff
@@ -43,33 +43,53 @@ export const makeAgoricWalletConnection = async (chainStorageWatcher, rpc) => {
       }),
     );
 
+    let isFinished = false;
+    const watchUpdates = async () => {
+      const iterator = subscribeLatest(walletNotifiers.walletUpdatesNotifier);
+      for await (const update of iterator) {
+        if (isFinished) {
+          // If tx fails, don't wait for status updates.
+          return;
+        }
+        if (update?.updated !== 'offerStatus' || update.status.id !== id) {
+          continue;
+        }
+        if (update.status.error !== undefined) {
+          isFinished = true;
+          onStatusChange({ status: 'error', data: update.status.error });
+          return;
+        }
+        if (update.status.numWantsSatisfied === 0) {
+          isFinished = true;
+          onStatusChange({ status: 'refunded' });
+          return;
+        }
+        if (update.status.numWantsSatisfied === 1) {
```

Comment:
what about multiples? I think this would be future proof and safe enough:
```suggestion
        if (update.status.numWantsSatisfied > 0) {
```